serverless/virus-scanner/src/s3.service.ts

```diff
@@ -0,0 +1,182 @@
+import {
+  CopyObjectCommand,
+  DeleteObjectCommand,
+  GetObjectCommand,
+  S3Client,
+} from '@aws-sdk/client-s3'
+import pino from 'pino'
+
+import {
+  GetDeleteS3FileStreamParams,
+  GetS3FileStreamResult,
+  MoveS3FileParams,
+} from './types'
+
+export class S3Service {
+  private readonly s3Client: S3Client
+
+  constructor(isDevelopmentEnv: boolean, private readonly logger: pino.Logger) {
+    if (isDevelopmentEnv) {
+      this.s3Client = new S3Client({
+        region: 'ap-southeast-1',
+        endpoint: `http://host.docker.internal:4566`,
+        forcePathStyle: true,
+        credentials: {
+          accessKeyId: '',
+          secretAccessKey: '',
+        },
+      })
+    } else {
+      // lambda function should automatically pick configs at runtime in non-dev envs
+      this.s3Client = new S3Client({
+        region: 'ap-southeast-1',
+      })
+    }
+  }
+
+  async getS3FileStreamWithVersionId({
+    bucketName,
+    objectKey,
+  }: GetDeleteS3FileStreamParams): Promise<GetS3FileStreamResult> {
+    this.logger.info(
+      {
+        bucketName,
+        objectKey,
+      },
+      'Getting document from s3',
+    )
+
+    try {
+      const { Body: body, VersionId: versionId } = await this.s3Client.send(
+        new GetObjectCommand({
+          Key: objectKey,
+          Bucket: bucketName,
+        }),
+      )
+
+      if (!versionId) {
```

Comment:
should we check for empty body before proceeding?

Reply:
[2402958](https://github.com/opengovsg/FormSG/pull/6642/commits/240295823b2974aecf6712f48d99fdee0ad7ca3e)